ui/packages/consul-ui/app/components/collapsible-notices/index.js

```diff
@@ -1,3 +1,16 @@
 import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
 
-export default class CollapsibleNotices extends Component {}
+export default class CollapsibleNotices extends Component {
+  // =attributes
+  @tracked collapsible = false;
+
+  @action
+  countNotices() {
+    const allNotices = [
+      ...document.querySelectorAll('div.collapsible-notices > div.notices > .notice'),
+    ];
+    this.collapsible = allNotices.length > 2;
```

Comment:
In upcoming Topology changes we'll be adding new more notices. These code changes are a more efficient solution to showing or not the collapsible button.

Reply:
Curious as to why this could be more efficient? Mostly in modern frameworks the idea is to declaratively configure what you want the view to be based on the state and then the framework efficiently creates that view for you - using querySelector[All] is generally shied away from. I would say that generally if you find yourself using DOM methods like this it might be worth thinking of another way to do it (there's always an outlier but generally the reason for that outlier is clear if there is)